src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -914,6 +929,14 @@ private void AddAllocatedEndpointInfo(IEnumerable<AppResource> resources)
 
         foreach (var appResource in resources)
         {
+            if (appResource.DcpResource is Container container)
+            {
+                if (container.Spec.Networks?.Any() == true)
+                {
+                    containerHost = appResource.ModelResource.Name;
```

Comment:
This is the thing I'm not sure about. Container host is used by code to talk to host resources from a container. We need a different property for this.

Reply:
For executable resources, the endpoint container host should still be set to `host.docker.internal` (or equivalent), I only override to the container alias if the endpoint provider is a container (and the container is attached to a custom network, which probably isn't necessary with the latest changes).

Reply:
I'm not sure about this, if you try to make a "self" address that another resource can use to talk to you, it'll be incorrect. The resolution for host depends on the types of both resources.

Reply:
The logic I'm aiming for is that `ContainerHost`/`ContainerPort` are used if the requesting component is a container resource, but the value for each property will depend on the resource that's providing the endpoint.

| Endpoint Provider | Host Value | ContainerHost Value | ContainerPort Value |
| ------------------ | ---------- | ---------------- | --------------------- |
| Executable             | 127.0.0.1    |  "host.docker.internal" |  `Port`               |
| Container               | 127.0.0.1    | Network alias name | `TargetPort` |

The main caveat of the design is that executable resources are expected to use `Host` and `Port` to access services and containers to use `ContainerHost` and `ContainerPort`. I'll admit it's a bit of a kludge to force separate endpoint properties for container and non-container resources.

It might be interesting (long term) to think of the resources and endpoints more like a graph than just the endpoints being metadata on individual resources. The resources are nodes and the endpoints are directional edges between the nodes. Instead of querying a given resources for its endpoint metadata/annotations, we'd start the query from the consuming resource with the destination endpoint as the target. For example, `redisCommanderResource.FindServiceConnection(redisInstance, "tcp");` to get the endpoint/connection info to connect to a redis container from a redis commander sidecar.

Reply:
The problem is a component today that is using container host + port. That will start exploding right? Right now we assume a single network (the host network). We also want to handle this logic that replaces localhost with the container host https://github.com/dotnet/aspire/blob/89fdd219ac472f6151f9f4037b41dccbb2ea1063/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs#L1337

Reply:
Based on our conversation, I switched back to the temporary workaround approach for our sidecar resources. We'll hardcode the target resource container host name/target port and spend some time discussing what new service discovery API for these scenarios should look like before we make any public API changes.